scripts/remove-redundant-flags.js

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env node
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+'use strict';
+const fs = require('fs');
+const path = require('path');
+const { platform } = require('os');
+
+const compareVersions = require('compare-versions');
+
+const bcd = require('..');
+
+/** Determines if the OS is Windows */
+const IS_WINDOWS = platform() === 'win32';
+
+let twoYearsAgo;
+if (
+  process.env.npm_lifecycle_event &&
+  process.env.npm_lifecycle_event.includes('mocha')
+) {
+  // When running test, use a specific time for consistent testing
+  twoYearsAgo = new Date('January 1, 2019 0:00:00');
+} else {
+  twoYearsAgo = new Date(Date.now());
+  twoYearsAgo.setFullYear(twoYearsAgo.getFullYear() - 2);
+}
+
+const getEarliestVersion = (...args) => {
+  const versions = args
+    .filter(version => typeof version === 'string')
+    .map(version => version.replace('≤', ''));
+
+  let earliestVersion = versions[0];
+
+  for (const version of versions) {
+    if (
+      earliestVersion === 'preview' ||
+      (version !== 'preview' &&
+        compareVersions.compare(earliestVersion, version, '>'))
+    )
+      earliestVersion = version;
+  }
+
+  return earliestVersion;
+};
+
+const removeRedundantFlags = (key, value, limitBrowser) => {
+  if (key === '__compat') {
+    for (const [browser, rawSupportData] of Object.entries(value.support)) {
+      if (limitBrowser && browser != limitBrowser) continue;
+
+      if (rawSupportData !== undefined) {
+        const supportData = Array.isArray(rawSupportData)
+          ? rawSupportData
+          : [rawSupportData];
+        const result = [];
+
+        const simpleStatement = supportData.find(statement => {
```

Comment:
This code felt familiar, looks like this bit from the collector:
https://github.com/foolip/mdn-bcd-collector/blame/4488e1171a2bd83bfe40459aa27add2332449cfe/update-bcd.js#L276-L282

But I don't understand the purpose of looking for a simple statement here, why does this matter to flag removal?

It would be great write in prose as docs for this function what the rules are for when flag data is going to be removed. I think it should be something like this: For each flag entry, determine if there is a later non-flag entry and if the `version_added` of that entry is two years old or more. Such flag entries are candidates for removal. Then, if all of the flag entries are candidates and they are all at end of the array, remove them. Otherwise leave them, as removing them would leave a hole between non-flag entries, which the guideline carves out an exception for.